### PR TITLE
Service parallel fetch

### PR DESCRIPTION
**Pull request by mb-dbc-dk**
Support structure for fetching form rawrepo-oai-formatter-service in parallel

### CODE REVIEW

File: service/src/main/java/dk/dbc/rr/oai/Config.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
+/*
```
Comment (noah-dbc) [inline service/src/main/java/dk/dbc/rr/oai/Config.java]: this seems more or less a repetition of the config class from the formatter?

File: service/src/main/java/dk/dbc/rr/oai/Config.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        private FromEnv(String name, String value) {
+            this.name = name;
+            this.value = value;
+        }
+
+        private String get() {
+            return value;
+        }
+
+//        private FromEnv is(String test, Predicate<String> p) {
```
Comment (noah-dbc) [inline service/src/main/java/dk/dbc/rr/oai/Config.java]: get rid of this if it is not needed

File: service/src/main/java/dk/dbc/rr/oai/fetch/ParallelFetch.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    public URI buildUri(String id, String format, String sets) {
+        return config.getFormatServiceUri()
+                .queryParam("id", id)
+                .queryParam("format", format)
+                .queryParam("sets", sets)
+                .build();
+    }
+
+    /**
+     * Produces a list or of DOM Elements for the given uris
```
Comment (noah-dbc) [inline service/src/main/java/dk/dbc/rr/oai/fetch/ParallelFetch.java]: remove 'or' here?

File: service/src/main/java/dk/dbc/rr/oai/fetch/ParallelFetch.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        log.info("Requesting {} uris", uris.size());
+        threads.clear();
+        ExecutorService executor;
+        executor = Executors.newFixedThreadPool(config.getParallelFetch());
+        List<Future<Document>> requests = uris.stream()
+                .map(u -> executor.submit(() -> fetcher(u)))
+                .collect(toList());
+        executor.shutdown();
+        try {
+            System.out.println("threads = " + threads);
```
Comment (noah-dbc) [inline service/src/main/java/dk/dbc/rr/oai/fetch/ParallelFetch.java]: sure you want to leave a sysout there?

File: service/src/main/java/dk/dbc/rr/oai/Config.java (hunk 1 above)
Comment (mb-dbc-dk) [reply service/src/main/java/dk/dbc/rr/oai/Config.java]: Yup. Just feels like an overkill to put one class in it's own module. But there's only so many ways to extract config ;-)

File: service/src/main/java/dk/dbc/rr/oai/Config.java (hunk 2 above)
Comment (mb-dbc-dk) [reply service/src/main/java/dk/dbc/rr/oai/Config.java]: It will eventually, but not really rocket science to write again.

**Pull request merged**